debezium-core/src/main/java/io/debezium/transforms/UnwrapFromEnvelope.java

```diff
@@ -0,0 +1,113 @@
+ *
+ * @param <R> the subtype of {@link ConnectRecord} on which this transformation will operate
+ * @author Jiri Pechanec
+ */
+public class UnwrapFromEnvelope<R extends ConnectRecord<R>> implements Transformation<R> {
+
+    private static final String ENVELOPE_SCHEMA_NAME_SUFFIX = ".Envelope";
+    private final Logger logger = LoggerFactory.getLogger(getClass());
+    private static final Field DROP_INCOMING_TOMBSTONES = Field.create("drop.incoming.tombstones")
+            .withDisplayName("Drop incoming tombstones")
+            .withType(ConfigDef.Type.BOOLEAN)
+            .withWidth(ConfigDef.Width.SHORT)
+            .withImportance(ConfigDef.Importance.LOW).withDefault(true)
+            .withDescription("Debezium by default generates a tombstone record to enable Kafka compaction after "
+                    + "a delete record was generated. This record is usually filtered out to avoid duplicates "
+                    + "as a delete record is converted to a tombstone record too");
+
+    private static final Field DROP_OUTGOING_TOMBSTONES = Field.create("drop.outgoing.tombstones")
+            .withDisplayName("Drop outgoing tombstones")
+            .withType(ConfigDef.Type.BOOLEAN)
+            .withWidth(ConfigDef.Width.SHORT).withImportance(ConfigDef.Importance.MEDIUM)
+            .withDefault(false)
+            .withDescription("Drop tombstones records if a processiong connector cannot process them or "
+                    + "a compaction is undesirable.");
+
+    private boolean dropIncomingTombstones;
+    private boolean dropOutgoingTombstones;
+    private ExtractField<R> delegate = new ExtractField.Value<R>();
+
+    @Override
+    public void configure(final Map<String, ?> configs) {
+        final Configuration config = Configuration.from(configs);
+        final Field.Set configFields = Field.setOf(DROP_INCOMING_TOMBSTONES, DROP_OUTGOING_TOMBSTONES);
+        if (!config.validateAndRecord(configFields, logger::error)) {
+            throw new ConnectException("Unable to validate config.");
+        }
+
+        dropIncomingTombstones = config.getBoolean(DROP_INCOMING_TOMBSTONES);
+        dropOutgoingTombstones = config.getBoolean(DROP_OUTGOING_TOMBSTONES);
+
+        final Map<String, String> delegateConfig = new HashMap<>();
+        delegateConfig.put("field", "after");
+        delegate.configure(delegateConfig);
+    }
+
+    @Override
+    public R apply(final R record) {
+        if (record.value() == null) {
+            if (dropIncomingTombstones) {
+                logger.trace("Tombstone {} arrived and requested to be dropped", record.key());
+                return null;
+            }
+            return record;
+        }
+        if (!record.valueSchema().name().endsWith(ENVELOPE_SCHEMA_NAME_SUFFIX)) {
+            logger.warn("Expected Envelope for transformation, passing it unchanged");
+            return record;
+        }
+        final R newRecord = delegate.apply(record);
```

Comment:
Good idea, still I'd prefer to merge this one as it already adds some value - see demo https://github.com/jpechane/debezium-utils/tree/master/sink-demo
and open a follow-up Jira with this extension?